timely/src/progress/reachability.rs

```diff
@@ -156,11 +157,15 @@ impl<T: Timestamp> Builder<T> {
     /// Add links internal to operators.
     ///
     /// This method overwrites any existing summary, instead of anything more sophisticated.
-    pub fn add_node(&mut self, index: usize, inputs: usize, outputs: usize, summary: Vec<Vec<Antichain<T::Summary>>>) {
+    pub fn add_node(&mut self, index: usize, inputs: usize, outputs: usize, summary: Connectivity<T::Summary>) {
 
         // Assert that all summaries exist.
         debug_assert_eq!(inputs, summary.len());
-        for x in summary.iter() { debug_assert_eq!(outputs, x.len()); }
+        for x in summary.iter() { 
+            for o in x.keys() {
+                debug_assert!(o < &outputs);
+            }
+        }
```

Comment:
I wonder if Rust is smart enough to eliminate the loop if the body is empty. I haven't seen the old version of the loop ever stand out, so probably not an issue?

Reply:
I'd guess it does, but that's a good point that this could be written differently (and is, elsewhere in the code). Brb!